server/graphql.server.ts

```diff
@@ -1,22 +1,63 @@
 import {PropertiesHolder} from '../properties/propertiesHolder';
 import {PolarisProperties} from '../properties/polarisProperties';
-import {Config, makeExecutableSchema} from 'apollo-server';
+import {makeExecutableSchema} from 'apollo-server';
 import {ApolloServer} from 'apollo-server-express';
 import {PolarisRequestHeaders} from "../http/request/polarisRequestHeaders";
+import {ApplicationLogProperties} from "@enigmatis/polaris-logs"
+import {GraphQLLogProperties} from "../logging/GraphQLLogProperties";
+import {InjectableLogger} from "../logging/GraphQLLogger";
+import {LogProperties} from "../properties/LogProperties";
+import {provide, buildProviderModule} from "inversify-binding-decorators";
+import {Container, inject} from "inversify";
+import {ISchemaCreator} from "../schema/utils/schema.creator";
 
 const path = require('path');
 const express = require('express');
 const app = express();
 
+export interface IPolarisGraphQLServer {
+    start();
+}
+
+@provide("IPolarisGraphQLServer")
 export class PolarisGraphQLServer {
     private server: ApolloServer;
     private _polarisProperties: PolarisProperties;
+    private _logProperties: LogProperties;
+    private polarisLogger: InjectableLogger;
 
-    constructor(config: Config) {
-        let executableSchema = makeExecutableSchema(config.schema);
-        let propertiesPath = path.join('../', "properties.json");
-        PropertiesHolder.loadPropertiesFromFile(propertiesPath);
+    constructor(//@inject("InjectableLogger")polarisLogger :InjectableLogger,
+               // @inject("ISchemaCreator")creator :ISchemaCreator
+    ) {/*
+            //this.polarisLogger = polarisLogger;
+            let schema = creator.generateSchema()
+            let executableSchemaDefinition: { typeDefs: any, resolvers: any } = {
+                typeDefs: schema.def,
+                resolvers: schema.resolvers
+            };
+            let executableSchema = makeExecutableSchema(executableSchemaDefinition);*/
+        let config = { typeDefs:
+                [ 'schema {query: Query, mutation: Mutation}',
+                    'type Query {\n                    books: [Book] }',
+                    '\n             type Book implements CommonEntity {\n                 id: ID!\n                 creationDate: String,\n                 lastUpdateDate: String,\n                 dataVersion: Int!,\n                 title: String @upper,\n                 author: String,\n                 otherBook: Book\n             }\n         ',
```

Comment:
no no
there is a tag that allows you not to use `\n` and all this shit
it looks bad

and why at all we have a schema inside polaris??

Reply:
outdated